### PR TITLE
[Merged by Bors] - chore: bump std and aesop

### PR DESCRIPTION
**Pull request by eric-wieser**
This includes leanprover/std4#301 which requires slight tweaks to the `List.sublists` API.

One other proof also breaks, presumably due to other Std4 commits.

---

- [x] depends on: https://github.com/leanprover/std4/pull/301

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
bors d+

**Comment by semorrison** [comment]
bors p=10

bors merge

File: lake-manifest.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
   {"git":
    {"url": "https://github.com/leanprover-community/aesop",
     "subDir?": null,
-    "rev": "b601328752091a1cfcaebdd6b6b7c30dc5ffd946",
+    "rev": "9dc4a1097a690216eaa7cf2d2290efd447e60d7a",
```
Comment (eric-wieser) [inline lake-manifest.json:23]: Is this intended? Is this a bump or a downgrade?

**Comment by digama0** [comment]
Note: I have pushed an update to the lean toolchain because this PR was on a [buggy version](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Potential.20data.20loss.20from.20.60lake.20clean.60.20with.204.2E2.2E0-rc2.2F3/near/397875701) of the toolchain. **WARNING**: checking out old commits of this PR using v4.2.0-rc2 or v4.2.0-rc3 can cause `lake clean` to delete your mathlib folder! If you need to do so, make sure to delete `lakefile.olean` manually before running any `lake` commands.

File: lake-manifest.json (hunk 1 above)
Comment (semorrison) [reply lake-manifest.json:23]: Yes, I ran `lake update` again.

**Comment by semorrison** [comment]
I've fixed another proof.

This PR appears to be holding up getting `nightly-testing` working again, which is then causing https://github.com/leanprover/lean4/pulls?q=is%3Apr+is%3Aopen+label%3Abreaks-mathlib to pile up.

If someone doesn't like my `ring_nf; simp_all; ring_nf` proof, then can we please fix it in a follow-up PR.

**Comment by semorrison** [comment]
bors merge

**Comment by eric-wieser** [comment]
bors merge

The PR description and title was no longer accurate

**Pull request closed without merging**